packages/typespec-ts/src/transform/transformApiVersionInfo.ts

```diff
@@ -26,23 +27,20 @@ export function transformApiVersionInfo(
   urlInfo?: UrlInfo
 ): ApiVersionInfo | undefined {
   const program = dpgContext.program;
-  const queryVersionDetail = getOperationQueryApiVersion(
+  const queryVersionDetail = getOperationApiVersion(
     client,
     program,
     dpgContext
   );
   const pathVersionDetail = extractPathApiVersion(urlInfo);
   const isCrossedVersion =
     pathVersionDetail?.isCrossedVersion || queryVersionDetail?.isCrossedVersion;
-  let defaultValue =
-    getEnrichedDefaultApiVersion(program, dpgContext) ??
-    pathVersionDetail?.defaultValue ??
-    queryVersionDetail?.defaultValue;
-
-  // Clear the default value if there exists cross version
-  if (isCrossedVersion) {
-    defaultValue = undefined;
-  }
+  const defaultValue =
+    (pathVersionDetail || queryVersionDetail) && !isCrossedVersion
+      ? getEnrichedDefaultApiVersion(program, dpgContext) ??
```

Comment:
I found there is inconsistency cross languages with default value and i am checking wiht tsp team [here](https://teams.microsoft.com/l/message/19:906c1efbbec54dc8949ac736633e6bdf@thread.skype/1704709079749?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1676977636162&teamName=Azure%20SDK&channelName=TypeSpec%20Discussion%20%F0%9F%90%AE&createdTime=1704709079749).

Reply:
Giving what Mark and Timothee mentioned in the channel, I think it's align with what we have in our implementation, also if there're cases that we should revisit our decision on the default value logic, we could file another PR to fix it, as this PR is trying to remove the extra api version in query parameter, what do you think ?